### DISCUSSION TITLE
Read email/attachment coming from a PEC address

### DISCUSSION THREAD

**Discussion by peakala** [ecederstrand/exchangelib#1309] [Q&A]
Hello,
when I try to access to a PEC email received into an outlook mailbox I'm able to retrieve a file that is called smime.p7m.
PEC (fw->) MAILBOX read by exchangelib

Somehow, if I forward this PEC from the PEC to another account and from that account to the one read by exchangelib, in such a case the script is able to read the email content and attachment as expected (as an .eml):
PEC (fw->) MAILBOX A (fw->) MAILBOX read by exchangelib

Is there a way to obtain the result as in the second way but executing just one forward as in the first way?

I've read this but I found no suitable solutions
https://github.com/ecederstrand/exchangelib/discussions/1132

Thanks

**Comment by ecederstrand** [comment]
I don't know how Exchange handles PEC email, som I'm afraid I can't help you.

Are you doing the forwarding with exchangelib? If you're doing it with Outlook, then probably Outlook is doing the decryption before forwarding. exchangelib itself does not provide any support for encrypted or certified emails beyond accessing the smime.p7m file.

**Reply by peakala** [reply]
Clear. Thanks.

I am forwarding it manually.

Do you know any suitable way to unencrypt the p7m file?

**Reply by ecederstrand** [reply]
I haven't tried unencrypting myself. There are some suggestions in https://github.com/ecederstrand/exchangelib/discussions/1132 but I haven't heard if they work.

**Reply by peakala** [reply]
It turned out that the p7m attachment read by exchangelib in the signed email is a "Content-Type: multipart/signed; protocol="application/pkcs7-signature"; micalg=sha-256; " **and not crypted**.

Thus, one might use a mail parser directly, like this:
```python
import email
msg = email.message_from_bytes(attachment.content)

attachments = []
for part in msg.walk():
    part_header = part.get('Content-Disposition')
````

The parts contain both the ".eml" component and its nested ".pdf"... meaning that this basic procedure is not able to see this email with this hiearchical structure (p7m contains the .eml which contains the pdf).

Question... is there a way to cast this p7m to the exchangelib's message class, so that I can reuse code parts?

Thank you

**Reply by ecederstrand** [reply]
Thanks for the update!

exchangelib does not support creating a `Message` object from MIME content or other formats. You would have to create your own method to do that.
